homeassistant/components/rdw/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Support for RDW binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable
+
+from vehicle import Vehicle
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import CONF_LICENSE_PLATE, DOMAIN, ENTRY_TYPE_SERVICE
+
+
+@dataclass
+class RDWBinarySensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    is_on_fn: Callable[[Vehicle], bool | None]
+
+
+@dataclass
+class RDWBinarySensorEntityDescription(
+    BinarySensorEntityDescription, RDWBinarySensorEntityDescriptionMixin
+):
+    """Describes RDW binary sensor entity."""
+
+
+BINARY_SENSORS: tuple[RDWBinarySensorEntityDescription, ...] = (
+    RDWBinarySensorEntityDescription(
+        key="liability_insured",
+        name="Liability Insured",
+        icon="mdi:shield-car",
+        is_on_fn=lambda vehicle: vehicle.liability_insured,
```

Comment:
Why have an `is_on_fn`, seems like it is just fetching the key as an attribute for both?

Reply:
Using getattr can be done, however has a typing disadvantage (as getattr returns Any).